common/timer.h

```diff
@@ -50,7 +50,7 @@ class Timer {
     }
   }
 
- private:
+ protected:
```

Comment:
do we have to make it protected? anything we want to use in it's extended class.